feat: [CDS-86606]: add best practices for expressions usage (#4620)
harness/developer-hub: docs/platform/variables-and-expressions/harness-variables.md

For example, `<+<+pipeline.name> == "pipeline1">` or `<+<+stage.variables.v1> == "dev">`.

> > Note: For expressions of type string, greater than or less than operators are not supported. Only equals and not equals are supported.

Reviewer: ```suggestion
:::info note
Greater than and Less than operators are not supported for string expression types. Only Equal to and Not equal to are supported.

:::
```

### Best practices for expressions usage

> - While using an expression, if we want it to be treated as a string, it should be wrapped within quotation marks.

Reviewer: ```suggestion
- When using an expression, if you want to treat it as a string, you just wrap it within quotation marks.
```

  
  For example, consider following expression:

  ```
  <+<+pipeline.variables.changeType> =~ ["<+stage.name>","All"]>
  ```
  

>   In the above expression, the expression `<+stage.name>` is wrapped within quotation marks because it is an element in a list of strings.

Reviewer: ```suggestion
  In the above expression, the `<+stage.name>` is wrapped within quotation marks because it is an element in a list of strings.
```

### Best practices for expressions usage

> - When using an expression, if you want to treat it as a string, you just wrap it within quotation marks.

Reviewer: ```suggestion
- When using an expression, if you want to treat it as a string, you must wrap it within quotation marks.
```
